build/stage-build-samples.yml

```diff
@@ -0,0 +1,124 @@
+jobs:
+- job: VS_Latest
+  timeoutInMinutes: 600
+
+  pool:
+    vmImage: windows-2022
+
+  steps:
+  - checkout: self
+    clean: true
+  
+  - powershell: |
+      if($env:System_PullRequest_TargetBranch -eq $null) { $TargetBranch = "master" } else { $TargetBranch = $env:System_PullRequest_TargetBranch; }
+
+      $dict = New-Object 'System.Collections.Generic.Dictionary[String,System.Collections.Generic.Dictionary[String,String]]'
+      $samples = Get-ChildItem -Path **\*.sln -Recurse | Where-Object {$_.FullName -notmatch "\\ArchivedProjects\\"}
+      foreach($sample in $samples){
+
+        $solutionPath=[System.IO.Path]::GetDirectoryName($sample.FullName);
+        Write-Host "$solutionPath"
+
+        git diff --quiet HEAD "origin/$TargetBranch" -- "$solutionPath"
+
+        # Only build if there a change in the solution path for the current PR, if we're not in a PR
+        if( ("$env:System_PullRequest_PullRequestId" -eq '') -or ($global:LASTEXITCODE -ne 0)) {
+          Write-Host "Adding $solutionPath"
+          $item = New-Object 'System.Collections.Generic.Dictionary[String,String]'
+          $item.Add("solutionPath", $sample.FullName)
+          $name = $sample.Name.Split(".")[0]
+          if(!$dict.ContainsKey($name)){
+            $dict.Add($name, $item)
+          }
+        }
+        $global:LASTEXITCODE = 0
+
+      }
+
+      $solutionsJson = $dict | ConvertTo-Json -Compress
+      
+      Write-Host $solutionsJson
```

Comment:
CI in this PR has built XamlBrewerUnoApp even though solutionsJson appears to be empty. Is that a problem? Not sure what could be causing it.

Reply:
@Youssef1313 I will double-check, I just mainly moved the previous code for the pipeline in here to build the samples. Maybe I have missed or change something

Reply:
@agneszitte Thanks. We can also cleanup in a separate PR if we want to get this one merged quickly!

Reply:
I finished reviewed the logic and made the proper adjustments :)